internal/core/docker_context_test.go

```diff
@@ -0,0 +1,55 @@
+package core
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// setupDockerContexts creates a temporary directory structure for testing the Docker context functions.
+// It creates the following structure, where $i is the index of the context, starting from 1:
+// - $HOME/.docker
+//   - config.json
+//   - contexts
+//   - meta
+//   - context$i
+//   - meta.json
+//
+// The config.json file contains the current context, and the meta.json files contain the metadata for each context.
+// It generates the specified number of contexts, setting the current context to the one specified by currentContextIndex.
+func setupDockerContexts(t *testing.T, currentContextIndex int, contextsCount int) {
+	configDir := filepath.Join(getHomeDir(), configFileDir)
+
+	err := createTmpDir(configDir)
```

Comment:
question: looks like this is just `os.MkdirAll`, could we use [T.TempDir](https://pkg.go.dev/testing#T.TempDir) which automatically cleans up instead and wouldn't trash the users config, if that understanding is correct?

Reply:
See my comment here: https://github.com/testcontainers/testcontainers-go/pull/2810#discussion_r1786128082

Wdyt if we dedicate a follow-up PR for simplifying the **unit** tests for the socket path resolution? Some would apply for reading the tc config.

Reply:
If this is expected to override the home dir perhaps the quickest thing would be to have this function do that so its very clear and can't be accidentally used without that and destroy someone config.

Reply:
See my comment https://github.com/testcontainers/testcontainers-go/pull/2810#discussion_r1799036874 about refining that in a follow-up PR